docs/reference/yaml.md

```diff
@@ -239,7 +239,60 @@ The meanings and formats of `limits` and `requests` may vary depending on whethe
 
 See docs for [Docker Swarm](https://docs.docker.com/config/containers/resource_constraints/) or for [Kubernetes](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#how-pods-with-r    esource-limits-are-run).
 
-### YAML - environment variable substitution
+## Configuration
+The configuration section allows you to define additional configuration that is global to the entire stack, currently this mostly impacts function build time options.
+
+### Templates
+The `templates` list allows you to define the information required to pull the templates for your functions.  This list of templates will automatically be pulled when you build your functions. When configured correctly, this allows you to completely build your functions with just `faas-cli build`.  Without this section, you must manually `faas-cli template pull <source>` _before_ you use `faas-cli build`.
+
+```yaml
+configuration:
+  templates:
+    - name: python3-http
+      source: https://github.com/openfaas-incubator/python-flask-template
+```
+
+### Copy
+The `copy` list allows you to define additional project paths that will be copied into your function's handler folder.
+
+```yaml
+configuration:
+  copy:
+    - ./common
+    - ./data
+    - ./models
+```
+
+Given the above configuration, the build folder for a python `hello` function would look like
+
+```sh
+build
+└── hello
+    ├── Dockerfile
+    ├── function
+    │   ├── __init__.py
+    │   ├── models
+    │   ├── ...
+    │   │   └── model_n
+    │   ├── data
+    │   ├── ...
+    │   │   └── data_set_n
+    │   ├── common
+    │   ├── ...
+    │   │   └── utils.py
+    │   ├── handler.py
+    │   └── requirements.txt
+    ├── index.py
+    ├── requirements.txt
+    └── template.yml
+```
+
+The CLI also has a related flag `--copy-extra`.  When this flag is used, the paths specified by the flag will be _merged_ into the list from the YAML.  This means it will extend, not replace, the values specified in the file.
```

Comment:
Call out where that related flag is:
"The CLI also has a related flag `--copy-extra`..."
to
"The CLI `build` command also has a related flag `--copy-extra`"